components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1373,6 +1422,14 @@ where
                 self.register_raft_gc_log_tick();
                 self.register_check_leader_lease_tick();
             }
+
+            if self.fsm.peer.force_leader && r == StateRole::Follower {
+                // for some reason, it's not leader anymore
```

Comment:
Like what?

Reply:
Don't know, it's unexpected.

Reply:
It is possible in the race condition that after the majority of the alive peer have removed all down peers and before the force leader received `SignificantMsg::ExitForceLeaderState` (i.e the force leader apply log slower than other peers or the force leader is isolated from PD), a new leader is elected and cause the force leader to step down.


 `ExitForceLeaderState` message is triggered by PD. After all peers on down stores are removed and before the message is received, there may have a new leader and cause this panic.

Reply:
Like network problem or raftstore stuck

Reply:
seems better to exit force leader here, and let's PD retry later